src/Microsoft.Diagnostics.Monitoring/DiagnosticsEventPipeProcessor.cs

```diff
@@ -289,6 +317,96 @@ private void PostMetric(Metric metric)
             }
         }
 
+        private Task HandleGCEvents(EventPipeEventSource source, int pid, Func<Task> stopFunc, CancellationToken token)
+        {
+            int gcNum = -1;
+
+            TaskCompletionSource<object> gcEventsStarted = new TaskCompletionSource<object>();
+            token.Register(() => gcEventsStarted.TrySetCanceled(token));
+
+            TaskCompletionSource<object> gcStopped = new TaskCompletionSource<object>();
+            token.Register(() => gcStopped.TrySetCanceled(token));
+
+            source.Clr.GCStart += data =>
+            {
+                if (data.ProcessID != pid)
+                {
```

Comment:
Did you ever figure what that type table flush was for in the original code?

Reply:
This is the reason for the type flush: https://github.com/microsoft/perfview/blob/6e9043ace41ed0917cf566f387f4a7cce5e26a57/src/EtwHeapDump/DotNetHeapDumper.cs#L35-L38

Preserved from PerfView, but it seems like it might only be necessary for desktop framework.  I haven't done any digging on whether we still need the extra flush of the type cache in modern .NET.